### PR TITLE
test: visual tests

### PR DESCRIPTION
**Pull request by niktverd**

### CODE REVIEW

File: .github/workflows/pr-playwright-report.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        id: pr
+        run: echo "::set-output name=id::$(<pr/pr-id.txt)"
+        shell: bash
+      - name: Upload
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.STORYBOOK_S3_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.STORYBOOK_S3_SECRET_KEY }}
+          AWS_DEFAULT_REGION: ru-central1
+          AWS_EC2_METADATA_DISABLED: true
+        run: aws s3 cp playwright-report s3://playwright-reports/page-constructor/pulls/${{ steps.pr.outputs.id }}/ --endpoint-url=https://storage.yandexcloud.net --recursive
```
Comment (NasgulNexus) [inline .github/workflows/pr-playwright-report.yml:30]: page-constructor -> blog-constructor

File: .github/workflows/pr-playwright-report.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        shell: bash
+      - name: Create Comment
+        uses: marocchino/sticky-pull-request-comment@v2
+        with:
+          GITHUB_TOKEN: ${{ secrets.GRAVITY_UI_BOT_GITHUB_TOKEN }}
+          number: ${{ steps.pr.outputs.id }}
+          header: playwright test
+          message: '[Playwright Test Component](https://storage.yandexcloud.net/playwright-reports/page-constructor/pulls/${{ steps.pr.outputs.id }}/index.html) is ready.'
```
Comment (NasgulNexus) [inline .github/workflows/pr-playwright-report.yml:38]: page-constructor -> blog-constructor

File: scripts/playwright-docker.sh [hunk 3]
```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+IMAGE_NAME="mcr.microsoft.com/playwright"
+IMAGE_TAG="v1.45.3-jammy" # This version have to be synchronized with playwright version from package.json
+
+NODE_MODULES_CACHE_DIR="$HOME/.cache/page-constructor-playwright-docker-node-modules"
```
Comment (NasgulNexus) [inline scripts/playwright-docker.sh:8]: page-constructor -> blog-constructor

File: playwright/core/expectScreenshotFixture.ts [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            const promises = locators.map((locator) =>
+                locator.evaluate(
+                    (image: HTMLImageElement) =>
+                        image.complete ||
+                        new Promise<unknown>((resolve) => image.addEventListener('load', resolve)),
+                ),
+            );
+            await Promise.all(promises);
+
+            if (waitingFunctionLocal && isFunction(waitingFunctionLocal)) {
```
Comment (NasgulNexus) [inline playwright/core/expectScreenshotFixture.ts:42]: it seems not necessary

File: src/blocks/Author/__tests__/Author.visual.test.tsx [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+import {test} from '../../../../playwright/core/index';
+
+import {Default} from './helpers';
+
+test.describe('Author', () => {
+    test('render stories <Default>', async ({mount, expectScreenshot, defaultDelay}) => {
+        await mount(<Default />);
+        await defaultDelay();
```
Comment (NasgulNexus) [inline src/blocks/Author/__tests__/Author.visual.test.tsx:10]: await defaultDelay(); is it really necessary?

**Review by NasgulNexus — CHANGES_REQUESTED** [review]

File: src/blocks/Author/__tests__/Author.visual.test.tsx (hunk 5 above)
Comment (niktverd) [reply src/blocks/Author/__tests__/Author.visual.test.tsx:10]: I think it is necessary

File: playwright/core/expectScreenshotFixture.ts (hunk 4 above)
Comment (niktverd) [reply playwright/core/expectScreenshotFixture.ts]: sure

**Review by NasgulNexus — DISMISSED** [review]

**Review by NasgulNexus — APPROVED** [review]

**Review by NikitaCG — APPROVED** [review]

**Pull request merged**